### PR TITLE
feat: Vehicle has permission

### PR DESCRIPTION
**Pull request by rishabluthra**
Added vehicle has permissions to check if a vehicle contains a certain permission in prep for Connect Single Select.

### CODE REVIEW

File: lib/vehicle.js [hunk 1]
```diff
@@ -91,6 +91,24 @@ Vehicle.prototype.permissions = function() {
     });
 };
 
+/**
+ * Checks if vehicle's permission contains specified permission.
```
Comment (sydneypalumbo) [inline lib/vehicle.js]:
```suggestion
 * Checks if permissions granted to a vehicle contain the specified permission(s).
```

File: lib/vehicle.js [hunk 2]
```diff
@@ -91,6 +91,24 @@ Vehicle.prototype.permissions = function() {
...
+ *
+ * @param {String[]|String} permissions Permission to check on vehicle.
+ *
+ * @return {Promise.<String[]>} An array of permissions names.
```
Comment (sydneypalumbo) [inline lib/vehicle.js]:
```suggestion
 * @return {Promise.<Boolean>} Whether the vehicle has the specified permissions
```

File: lib/vehicle.js [hunk 3]
```diff
@@ -91,6 +91,24 @@ Vehicle.prototype.permissions = function() {
     });
 };
 
+/**
+ * Checks if vehicle's permission contains specified permission.
+ *
+ * @param {String[]|String} permissions Permission to check on vehicle.
```
Comment (sydneypalumbo) [inline lib/vehicle.js]:
```suggestion
 * @param {String[]|String} permissions Permission(s) to check if the vehicle has
```

File: test/unit/lib/vehicle.js [hunk 4]
```diff
@@ -132,6 +132,33 @@ test('permissions', async function(t) {
...
+    .reply(200, {
+      permissions: ['permission1', 'permission2', 'permission3'],
+    });
+
+  const hasPermissions = await vehicle.hasPermissions(
+    ['permission1', 'permission2']);
+
+  t.is(hasPermissions, true);
+});
+
```
Comment (sydneypalumbo) [inline test/unit/lib/vehicle.js]: can you add a test to check false cases as well :smile_cat:

**Comment by sydneypalumbo** [comment]
also don't forget to update the docs!

File: lib/vehicle.js (hunk 3 above)
Comment (rishabluthra) [reply lib/vehicle.js]: :neckbeard:

File: lib/vehicle.js [hunk 5]
```diff
@@ -91,6 +91,29 @@ Vehicle.prototype.permissions = function() {
     });
 };
 
+/**
+ * Checks if permissions granted to a vehicle contain the specified permission(s).
+ *
+ * @param {String[]|String} permissions Permission(s) to check if the vehicle has
+ *
+ * @return {Promise.<Boolean>} An array of permissions names.
```
Comment (sydneypalumbo) [inline lib/vehicle.js]:
```suggestion
 * @return {Promise.<Boolean>} Whether the vehicle has the specified permission(s) 
```

File: lib/vehicle.js [hunk 6]
```diff
@@ -91,6 +91,29 @@ Vehicle.prototype.permissions = function() {
...
+ */
+Vehicle.prototype.hasPermissions = function(permissions) {
+  return this.permissions()
+    .then(function(vehiclePermissions) {
+      if (_.isArray(permissions)) {
+        const strippedPermissions = _.map(permissions, (permission) => {
```
Comment (sydneypalumbo) [inline lib/vehicle.js]: nit: could do a simple return arrow function
```
const strippedPermissions = _.map(permissions, (permission) =>
  permssion.replace(/^required:/, '');
);
```

**Pull request merged**